src/transformers/adapters/peft_mixin.py

```diff
@@ -0,0 +1,333 @@
+                revision argument to be passed to `hf_hub_download` method from `huggingface_hub`.
+            use_auth_token (`str`, `optional`):
+                use_auth_token argument to be passed to `hf_hub_download` method from `huggingface_hub`.
+            commit_hash (`str`, `optional`):
+                commit_hash argument to be passed to `hf_hub_download` method from `huggingface_hub`.
+            device_map (`str`, `optional`):
+                The device map to use for the adapter. Used only in the case the model is offloaded into the CPU to
+                correctly re-dispatch the model together with the adapter weights.
+            max_memory (`str`, `optional`):
+                The maximum memory to use for the adapter. Used only in the case the model is offloaded into the CPU
+                for the same reasons stated above.
+            offload_dir (`str`, `optional`):
+                The directory to use for offloading the model. Used only in the case the model is offloaded into the
+                CPU.
+            offload_index (`int`, `optional`):
+                `offload_index` argument to be passed to `accelerate.dispatch_model` method.
+        """
+        check_peft_version(min_version="0.4.0")
+
+        adapter_name = adapter_name or "default"
+
+        from peft import PeftConfig, inject_adapter_in_model, load_peft_weights
+        from peft.utils import set_peft_model_state_dict
+
+        if not self._hf_peft_config_loaded:
+            self._hf_peft_config_loaded = True
+        elif adapter_name in self.peft_config:
+            raise ValueError(f"Adapter with name {adapter_name} already exists. Please use a different name.")
+
+        adapter_config_file = find_adapter_config_file(
+            peft_model_id,
+            revision=revision,
+            use_auth_token=use_auth_token,
+            commit_hash=commit_hash,
+        )
+
+        if adapter_config_file is None:
+            raise ValueError(
+                f"adapter model file not found in {peft_model_id}. Make sure you are passing the correct path to the "
+                "adapter model."
+            )
+
+        loaded_peft_config = PeftConfig.from_pretrained(
+            peft_model_id,
+            revision=revision,
+            use_auth_token=use_auth_token,
+            commit_hash=commit_hash,
+        )
+
+        # Replace the adapter with the loaded adapter
+        inject_adapter_in_model(loaded_peft_config, self, adapter_name)
+
+        adapter_state_dict = load_peft_weights(
+            peft_model_id,
+            revision=revision,
+            use_auth_token=use_auth_token,
+        )
+
+        # We need to pre-process the state dict to remove unneeded prefixes - for backward compatibility
+        processed_adapter_state_dict = {}
+        for key, value in adapter_state_dict.items():
+            if "base_model.model" in key:
+                new_key = key.replace("base_model.model.", "")
```

Comment:
Wondering if `"base_model.model"` should ever be anywhere else but at the start of the key? If not, would this not be more precise:

```python
            prefix = "base_model.model"
            if key.startswith(prefix):
                new_key = key[len(prefix):]
```

Reply:
Very nice, yes indeed!

Reply:
Added it in f310b33